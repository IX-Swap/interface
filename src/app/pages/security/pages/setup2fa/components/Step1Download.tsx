import React from 'react'
import { Typography, Grid, Link, Box } from '@material-ui/core'
import appStoreLogo from '../assets/appstore.png'
import googlePlayLogo from '../assets/googleplay.png'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step1Download = () => (
  <StepWrapper title='Install Authenticator App in Your Smartphone Device'>
    <Grid container direction='column' spacing={3} alignItems='center'>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          When you log in, you’ll need to enter a code generated by the Google
          Authenticator app or another authenitcator app
        </Typography>
      </Grid>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          Download an authentication app to your smartphone from the App Store
          (for iPhone) or Google Play (for android smartphones)
        </Typography>
      </Grid>
      <Grid item>
        <Box paddingY={4}>
          <Grid container justify='center' spacing={2}>
            <Grid item>
              <Link
                href='https://apps.apple.com/us/app/google-authenticator/id388497605'
                target='_blank'
              >
                <img height='40px' src={appStoreLogo} alt='App Store Logo' />
              </Link>
            </Grid>
            <Grid item>
              <Link
                href='https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en'
                target='_blank'
              >
                <img
                  height='40px'
                  src={googlePlayLogo}
                  alt='Google Play Logo'
                />
              </Link>
            </Grid>
          </Grid>
        </Box>
      </Grid>
      <Grid item xs={12} md={8} lg={6}>
        <Typography align='center' variant='body1'>
          Please click “NEXT” when you have finished the authenticator app
          installed in your smartphone
        </Typography>
      </Grid>
    </Grid>
  </StepWrapper>
)
