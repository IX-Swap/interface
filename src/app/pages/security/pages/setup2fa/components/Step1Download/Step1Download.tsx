import React from 'react'
import { Typography, Grid, Link } from '@mui/material'
import { StepWrapper } from 'app/pages/security/components/StepWrapper'
import { ReactComponent as AppStoreLogo } from 'app/pages/security/pages/setup2fa/assets/appstore.svg'
import { ReactComponent as GooglePlayLogo } from 'app/pages/security/pages/setup2fa/assets/googleplay.svg'
import TwoFaIcon from 'app/pages/security/pages/setup2fa/assets/security.png'
import { useStyles } from './Step1Download.styles'

export const Step1Download = () => {
  const classes = useStyles()

  return (
    <StepWrapper
      title={
        <>
          Install Authenticator App
          <br /> in Your Smartphone Device
        </>
      }
    >
      <Grid container justifyContent={'center'}>
        <Grid
          item
          container
          direction='column'
          alignItems='center'
          className={classes.container}
        >
          <Grid item>
            <Typography
              align='center'
              variant='body1'
              className={classes.firstText}
            >
              When you log in, you will need to enter a code generated by the
              Google Authenticator app or another authenticator app
            </Typography>
          </Grid>
          <Grid item>
            <img src={TwoFaIcon} alt={'Enabled 2FA'} className={classes.icon} />
          </Grid>
          <Grid item>
            <Typography
              align='center'
              variant='body1'
              className={classes.secondText}
            >
              Download an Authentication App to your smartphone from the App
              Store or Google Play
            </Typography>
          </Grid>

          <Grid item container justifyContent='center' gap={2.5}>
            <Grid item>
              <Link
                href='https://apps.apple.com/us/app/google-authenticator/id388497605'
                target='_blank'
              >
                <AppStoreLogo />
              </Link>
            </Grid>
            <Grid item>
              <Link
                href='https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en'
                target='_blank'
              >
                <GooglePlayLogo />
              </Link>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
