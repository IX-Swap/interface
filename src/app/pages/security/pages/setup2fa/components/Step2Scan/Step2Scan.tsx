import React from 'react'
import { Typography, Box, Grid, Link } from '@mui/material'
import useStyles from 'app/pages/security/pages/setup2fa/components/Step2Scan/Step2Scan.styles'
import { StepWrapper } from 'app/pages/security/components/StepWrapper'
import { TwoFaData } from 'app/pages/security/types'

export interface Step2ScanProps {
  twoFaData: TwoFaData | undefined
}

export const Step2Scan = ({ twoFaData }: Step2ScanProps) => {
  const classes = useStyles()

  return (
    <StepWrapper title='Scan This QR Code in Your Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Use the app to scan the barcode below or type in the key. If you
            don’t have a smartphone or need more help, please{' '}
            <Link href='https://investax.io/contact/'>contact us.</Link>
          </Typography>
        </Grid>
        <Grid item>
          {twoFaData !== undefined && (
            <Grid container justifyContent='center'>
              <Box pt={4} pb={3}>
                <Grid container justifyContent='center' alignItems='center'>
                  <Grid item>
                    <div
                      data-testid='store-image'
                      className={classes.image}
                      style={{
                        backgroundImage: `url('${twoFaData.image}')`,
                        marginBottom: 10
                      }}
                    />
                  </Grid>
                  <Grid item>
                    <Typography className={classes.label}>
                      Click the “add icon “ on the app to scan the barcode below
                      or type in the key.
                    </Typography>
                    <Typography variant='h5' className={classes.key}>
                      {twoFaData.key}
                    </Typography>
                  </Grid>
                </Grid>
              </Box>
            </Grid>
          )}
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please click “NEXT” when you have successfully scanned...
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
