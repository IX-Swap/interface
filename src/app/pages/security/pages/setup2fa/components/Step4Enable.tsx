import React from 'react'
import { Typography, Box, Grid } from '@mui/material'
import { useEnable2fa } from '../hooks/useEnable2fa'
import { Enable2faFormValues } from '../types'
import { enable2faFormValuesSchema } from '../validation'
import { Form } from 'components/form/Form'
import { Submit } from 'components/form/Submit'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'
import { EnableFormFields } from 'app/pages/security/pages/setup2fa/components/EnableFormFields'

const defaultValues = {
  otp: ''
}

export const Step4Enable = () => {
  const [enable2fa] = useEnable2fa()
  const onSubmit = async (values: Enable2faFormValues) => {
    await enable2fa(values)
  }

  return (
    <StepWrapper title='Enable Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please enter the 6 digit code from your authenticator app
          </Typography>
        </Grid>
        <Grid item>
          <Form
            defaultValues={defaultValues}
            validationSchema={enable2faFormValuesSchema}
            onSubmit={onSubmit}
            data-testid='form'
          >
            <EnableFormFields />
            <Box my={4} width='100%' textAlign='center'>
              <Submit>Enable</Submit>
            </Box>
          </Form>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Once enabled your authenticator you will be logged out and re-enter
            the OTP in order to access the platform
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  )
}
