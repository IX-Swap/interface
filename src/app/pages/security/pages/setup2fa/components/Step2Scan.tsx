import React from 'react'
import { Typography, Box, Grid, Link } from '@material-ui/core'
import useStyles from './Step2Scan.styles'
import { useSetup2fa } from '../hooks/useSetup2fa'
import { useSetup2faStore } from '../context'
import { useObserver } from 'mobx-react'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step2Scan = () => {
  const classes = useStyles()
  const store = useSetup2faStore()
  const { isLoading } = useSetup2fa()

  return useObserver(() => (
    <StepWrapper title='Scan This QR Code in Your Authenticator App'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Use the app to scan the barcode below or type in the key. If you
            don’t have a smartphone or need more help, please{' '}
            <Link href='https://investax.io/contact/'>contact us.</Link>
          </Typography>
        </Grid>
        <Grid item>
          {!isLoading && (
            <Grid container justifyContent='center'>
              <Box pt={4} pb={3}>
                <Grid container justifyContent='center' alignItems='center'>
                  <Grid item>
                    <div
                      data-testid='store-image'
                      className={classes.image}
                      style={{
                        backgroundImage: `url('${store.image}')`,
                        marginBottom: 10
                      }}
                    />
                  </Grid>
                  <Grid item>
                    <Typography className={classes.label}>
                      Click the “add icon “ on the app to scan the barcode below
                      or type in the key.
                    </Typography>
                    <Typography variant='h5' className={classes.key}>
                      {store.key}
                    </Typography>
                  </Grid>
                </Grid>
              </Box>
            </Grid>
          )}
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please click “NEXT” when you have sucessfully scanned...
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  ))
}
