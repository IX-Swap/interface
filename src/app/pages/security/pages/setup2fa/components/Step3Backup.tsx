import React from 'react'
import { Typography, Grid } from '@material-ui/core'
import useStyles from './Step3Backup.styles'
import { useSetup2faStore } from '../context'
import { useObserver } from 'mobx-react'
import { StepWrapper } from 'app/pages/security/pages/setup2fa/components/StepWrapper'

export const Step3Backup = () => {
  const classes = useStyles()
  const store = useSetup2faStore()

  return useObserver(() => (
    <StepWrapper title='Save This Backup Key'>
      <Grid container direction='column' spacing={3} alignItems='center'>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please save this key code in a safe place. It will allow you to
            recover your authenticator in case of losing your phone
          </Typography>
        </Grid>
        <Grid item>
          <Typography variant='h4' align='center' className={classes.key}>
            {store.key}
          </Typography>
        </Grid>
        <Grid item xs={12} md={8} lg={6}>
          <Typography align='center' variant='body1'>
            Please click “Next” if you have sucessfully stored this key.
          </Typography>
        </Grid>
      </Grid>
    </StepWrapper>
  ))
}
