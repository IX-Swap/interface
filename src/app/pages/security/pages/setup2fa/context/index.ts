import { Setup2faStore } from './store'
import generateStoreHookAndProvider from 'helpers/generateStoreHookAndProvider'

const store = new Setup2faStore()

export const {
  Provider: Setup2faProvider,
  useStore: useSetup2faStore
} = generateStoreHookAndProvider<Setup2faStore>(store)
