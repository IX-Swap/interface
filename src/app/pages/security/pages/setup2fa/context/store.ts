import { action, observable } from 'mobx'
import { TwoFaData } from 'app/pages/security/pages/setup2fa/types'

export class Setup2faStore {
  public steps = [
    'Download app',
    'Scan QR Code',
    'Backup Key',
    'Enable Google Authenticator'
  ]

  @observable
  public activeStep = 0

  @observable
  public image = ''

  @observable
  public key = ''

  @observable
  public encoded = ''

  @action
  public setActiveStep = (step = 0): void => {
    this.activeStep = step
  }

  @action nextPage = (): void => {
    this.activeStep = Math.min(this.activeStep + 1, this.steps.length - 1)
  }

  @action prevPage = (): void => {
    this.activeStep = Math.max(this.activeStep - 1, 0)
  }

  @action
  set2faData = (data: TwoFaData) => {
    this.image = data.image
    this.key = data.key
    this.encoded = data.encoded
  }
}
