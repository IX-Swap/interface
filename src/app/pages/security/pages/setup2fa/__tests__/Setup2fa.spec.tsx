import React from 'react'
import { render, cleanup } from 'test-utils'
import { Step1Download } from '../components/Step1Download'
import { Step2Scan } from '../components/Step2Scan'
import { Step3Backup } from '../components/Step3Backup'
import { Step4Enable } from '../components/Step4Enable'
import * as setupContext from '../context'
import { Setup2fa } from 'app/pages/security/pages/setup2fa/Setup2fa'
import { waitFor, fireEvent } from '@testing-library/react'
import { Setup2faStore } from '../context/store'

jest.mock('../components/Step1Download', () => ({
  Step1Download: jest.fn(() => null)
}))
jest.mock('../components/Step2Scan', () => ({
  Step2Scan: jest.fn(() => null)
}))
jest.mock('../components/Step3Backup', () => ({
  Step3Backup: jest.fn(() => null)
}))
jest.mock('../components/Step4Enable', () => ({
  Step4Enable: jest.fn(() => null)
}))

describe('Setup2fa', () => {
  const baseContext: Setup2faStore = {
    activeStep: 0,
    image: '',
    key: '',
    encoded: '',
    setActiveStep: jest.fn(),
    nextPage: jest.fn(),
    prevPage: jest.fn(),
    set2faData: jest.fn(),
    steps: [
      'Download app',
      'Scan QR Code',
      'Backup Key',
      'Enable Google Authenticator'
    ]
  }

  afterEach(async () => {
    await cleanup()
    jest.clearAllMocks()
  })

  it('renders without error', () => {
    render(<Setup2fa />)
  })

  it('renders Step1Download if activeStep is 0', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 0
    })

    render(<Setup2fa />)

    expect(Step1Download).toHaveBeenCalledTimes(1)
  })

  it('renders Step2Scan if activeStep is 1', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 1
    })

    render(<Setup2fa />)

    expect(Step2Scan).toHaveBeenCalledTimes(1)
  })

  it('renders Step3Backup if activeStep is 2', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 2
    })

    render(<Setup2fa />)

    expect(Step3Backup).toHaveBeenCalledTimes(1)
  })

  it('renders Step4Enable if activeStep is 3', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 3
    })

    render(<Setup2fa />)

    expect(Step4Enable).toHaveBeenCalledTimes(1)
  })

  it("renders Step1Download if activeStep doesn't match", () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 8
    })

    render(<Setup2fa />)

    expect(Step1Download).toHaveBeenCalledTimes(1)
  })

  it('renders Next button if step is not last', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 0
    })

    const { queryByRole } = render(<Setup2fa />)

    const nextButton = queryByRole('button')
    expect(nextButton).not.toBeNull()
  })

  it('does not render Next button if step is last', () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: baseContext.steps.length - 1
    })

    const { queryByRole } = render(<Setup2fa />)

    const nextButton = queryByRole('button')
    expect(nextButton).toBeNull()
  })

  it('invokes nextPage when Next button is clicked', async () => {
    jest.spyOn(setupContext, 'useSetup2faStore').mockReturnValue({
      ...baseContext,
      activeStep: 0
    })

    const { getByRole } = render(<Setup2fa />)

    const nextButton = getByRole('button')
    fireEvent.click(nextButton)
    await waitFor(() => {
      expect(baseContext.nextPage).toHaveBeenCalledTimes(1)
    })
  })
})
