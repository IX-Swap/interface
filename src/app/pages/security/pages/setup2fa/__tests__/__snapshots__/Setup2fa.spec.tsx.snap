// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup2fa should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-grid-xs-12 makeStyles-wrapper-1 css-9ot9es-MuiGrid-root"
  >
    <div
      class="MuiGrid-root makeStyles-leftBlock-2 css-vj1n65-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1vpwcmr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-2xfiep-MuiTypography-root"
            >
              Install Authenticator App
              <br />
               in Your Smartphone Device
            </h2>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column makeStyles-container-5 css-67pqv6-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-firstText-6 css-1gd5lqg-MuiTypography-root"
                  >
                    When you log in, you will need to enter a code generated by the Google Authenticator app or another authenticator app
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <img
                    alt="Enabled 2FA"
                    class="makeStyles-icon-8"
                    src="image-mock"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter makeStyles-secondText-7 css-1gd5lqg-MuiTypography-root"
                  >
                    Download an Authentication App to your smartphone from the App Store or Google Play
                  </p>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item css-di8adu-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-q3td2j-MuiTypography-root-MuiLink-root"
                      href="https://apps.apple.com/us/app/google-authenticator/id388497605"
                      target="_blank"
                    />
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-q3td2j-MuiTypography-root-MuiLink-root"
                      href="https://play.google.com/store/apps/details?id=com.google.android.apps.authenticator2&hl=en"
                      target="_blank"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item makeStyles-rightBlock-3 css-5x0lb4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-1c73l8u"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-93b3zb-MuiTypography-root"
            >
              Progress
            </p>
          </div>
          <div
            class="MuiStepper-root MuiStepper-vertical css-9tmuzo-MuiStepper-root"
          >
            <div
              class="MuiStep-root MuiStep-vertical css-tzsq05-MuiStep-root"
            >
              <button
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical active css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                  >
                    <div
                      class="makeStyles-wrapper-9 makeStyles-wrapper-12 MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-text-11 makeStyles-text-14 css-1b4l2pg-MuiTypography-root"
                      >
                        1
                      </p>
                    </div>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label Mui-active css-t1n505-MuiStepLabel-label"
                    >
                      Download App
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-tzsq05-MuiStep-root"
            >
              <button
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                  >
                    <div
                      class="makeStyles-wrapper-9 makeStyles-wrapper-15 MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-text-11 makeStyles-text-17 css-1b4l2pg-MuiTypography-root"
                      >
                        2
                      </p>
                    </div>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                    >
                      Scan QR Code
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-tzsq05-MuiStep-root"
            >
              <button
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                  >
                    <div
                      class="makeStyles-wrapper-9 makeStyles-wrapper-18 MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-text-11 makeStyles-text-20 css-1b4l2pg-MuiTypography-root"
                      >
                        3
                      </p>
                    </div>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                    >
                      Backup Key
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiStep-root MuiStep-vertical css-tzsq05-MuiStep-root"
            >
              <button
                class="MuiButtonBase-root MuiStepButton-root MuiStepButton-vertical css-rko9yk-MuiButtonBase-root-MuiStepButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiStepLabel-root MuiStepLabel-vertical css-1bl8ooz-MuiStepLabel-root"
                >
                  <span
                    class="MuiStepLabel-iconContainer css-4m07z1-MuiStepLabel-iconContainer"
                  >
                    <div
                      class="makeStyles-wrapper-9 makeStyles-wrapper-21 MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 makeStyles-text-11 makeStyles-text-23 css-1b4l2pg-MuiTypography-root"
                      >
                        4
                      </p>
                    </div>
                  </span>
                  <span
                    class="MuiStepLabel-labelContainer css-1n74cx2-MuiStepLabel-labelContainer"
                  >
                    <span
                      class="MuiStepLabel-label css-t1n505-MuiStepLabel-label"
                    >
                      Enable Authenticator
                    </span>
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container makeStyles-container-24 css-1bizp5f-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item makeStyles-item-25 css-13i4rnv-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root makeStyles-button-26 css-gbhykm-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Next
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
